src/models/NetworkAppController.js

```diff
@@ -161,9 +168,69 @@ export default class NetworkAppController {
     }
 
     // TODO: Check that package version matches the requested one
+    // TODO: Do not invoke setStdlib if matches existing one
     log.info(`Connecting to public deployment of ${this.package.stdlib.name} in ${this.network}`);
     const stdlibAddress = StdlibProvider.from(this.package.stdlib.name, this.network);
     await this.appManagerWrapper.setStdlib(stdlibAddress);
     this.networkPackage.stdlib = { address: stdlibAddress, ... this.package.stdlib };
   }
+
+  checkContracts(throwIfFail = false) {
+    const contracts = _.keys(this.package.contracts);
+    let msg;
+    
+    const [contractsDeployed, contractsMissing] = _.partition(contracts, (alias) => this.isContractDeployed(alias));
+    const contractsChanged = _.filter(contractsDeployed, (alias) => this.hasContractChanged(alias));
+
+    if (!_.isEmpty(contractsMissing)) {
+      msg = `Contracts ${contractsMissing.join(', ')} are not deployed.`;
+    } else if (!_.isEmpty(contractsChanged)) {
+      msg = `Contracts ${contractsChanged.join(', ')} have changed since the last deploy.`;
+    }
+
+    if (msg && throwIfFail) throw new Error(msg);
+    else if (msg) log.info(msg);    
+  }
+
+  checkContract(contractAlias, throwIfFail = false) {
+    let msg;
+    if (!this.isContractDefined(contractAlias)) {
+      msg = `Contract ${contractAlias} not found in application or stdlib`;
+    } else if (!this.isContractDeployed(contractAlias)) {
+      msg = `Contract ${contractAlias} is not deployed to ${this.network}.`;
+    } else if (this.hasContractChanged(contractAlias)) {
+      msg = `Contract ${contractAlias} has changed locally since the last deploy.`;
+    }
+
+    if (msg && throwIfFail) throw new Error(msg);
+    else if (msg) log.info(msg);
+  }
+
+  hasContractChanged(contractAlias) {
+    const contractName = this.package.contracts[contractAlias];
+    if (!this.isApplicationContract(contractAlias)) return false;
+    if (!this.isContractDeployed(contractAlias)) return true;
+    const contractClass = ContractsProvider.getFromArtifacts(contractName);
+    const currentBytecode = contractClass.bytecode;
+    const deployedBytecode = this.networkPackage.contracts[contractAlias].bytecode;
+    return currentBytecode != deployedBytecode;
+  }
+
+  isApplicationContract(contractAlias) {
+    return !!this.package.contracts[contractAlias];
+  }
+
+  isStdlibContract(contractAlias) {
+    if (!this.appController.hasStdlib()) return false;
+    const stdlib = new Stdlib(this.appController.package.stdlib.name);
+    return stdlib.hasContract(contractAlias);
+  }
+
+  isContractDefined(contractAlias) {
+    return this.isApplicationContract(contractAlias) || this.isStdlibContract(contractAlias);
+  }
+
+  isContractDeployed(contractAlias) {
+    return !this.isApplicationContract(contractAlias) || !_.isEmpty(this.networkPackage.contracts[contractAlias]);
+  }
```

Comment:
I'm starting to think that this controller is growing a lot and we may want to split it into a hierarchy in a short future

Reply:
Agree. But after EthBA :-P